mission-2/assets/js/app.js

```diff
@@ -0,0 +1,142 @@
+      const countryCurrencies = country.currencies;
+      if (countryCurrencies) {
+        Object.keys(countryCurrencies).forEach((currencyCode) => {
+          acc[currencyCode] = (acc[currencyCode] || 0) + 1;
+        });
+      }
+      return acc;
+    }, {});
+
+    return Object.entries(currencies)
+      .map(([currency, count]) => {
+        return `
+                    <tr>
+                        <td>${currency}</td>
+                        <td>${count}</td>
+                    </tr>
+                `;
+      })
+      .join("");
+  };
+
+  const renderCurrencyTable = (newHTML) => {
+    document.getElementById("currencies").innerHTML = newHTML;
+  };
+
+  const renderStatistics = (countries) => {
+    const totalCountries = countries.length;
+    const totalPopulation = countries.reduce(
+      (sum, country) => sum + country.population,
+      0
+    );
+    const averagePopulation = (totalPopulation / totalCountries).toFixed(2);
+
+    const generateStaticsTable = `
+        <h2>Statistics</h2> 
+            <tr>
+                <td>Total Countries:</td>
+                <td>${totalCountries}</td>
+            </tr>
+            <br>
+            <tr>
+                <td>Total Population:</td>
+                <td>${totalPopulation.toLocaleString()}</td>
+            </tr>
+            <br>
+            <tr>
+                <td>Average Population:</td>
+                <td>${averagePopulation}</td>
+            </tr>
+            <hr>
+        `;
+    document.getElementById("statistics").innerHTML = generateStaticsTable;
+  };
+
+  document.getElementById("buttonAll").addEventListener("click", async (e) => {
+    e.preventDefault();
+    try {
+      const data = await getData("https://restcountries.com/v3.1/all");
+      const countryTableHTML = generateCountryTable(data);
+      const regionTableHTML = generateRegionTable(data);
+      const currencyTableHTML = generateCurrencyStats(data);
+
+      renderCountryTable(countryTableHTML);
+      renderRegionTable(regionTableHTML);
+      renderStatistics(data);
+      renderCurrencyTable(currencyTableHTML);
+    } catch (error) {
+      console.warn(error);
+    }
+  });
+  // present input country
+  document
+    .getElementById("buttonInput")
+    .addEventListener("click", async (e) => {
+      e.preventDefault();
+      const userInput = document.getElementById("userInput").value.trim();
+      if (!userInput) return alert("Please enter a country name!");
+      try {
+        const data = await getData(
+          `https://restcountries.com/v3.1/name/${userInput}`
+        );
+        const countryTableHTML = generateCountryTable(data);
+        const regionTableHTML = generateRegionTable(data);
+        const currencyTableHTML = generateCurrencyStats(data);
+
+        renderCountryTable(countryTableHTML);
+        renderRegionTable(regionTableHTML);
+        renderStatistics(data);
+        renderStatistics(data);
```

Comment:
DRY! 
if you would have used a single function to handle both events, then the currencies would have worked right